pulpcore/cli/common/context.py

```diff
@@ -160,6 +161,20 @@ def wait_for_task(self, task: EntityDefinition, timeout: int = 120) -> Any:
         except KeyboardInterrupt:
             raise PulpNoWait(f"Task {task_href} sent to background.")
 
+    def _get_plugin_name_prior_to_3_11(self, name: str) -> Union[str, None]:
+        """The status API names changed in 3.11. Remove me in pulpcore==3.16"""
+        new_to_old_names_map: Dict[str, Union[str, None]] = defaultdict(lambda: None)
+        new_to_old_names_map.update(
+            dict(
+                ansible="pulp_ansible",
+                container="pulp_container",
+                core="pulpcore",
+                file="pulp_file",
+                rpm="pulp_rpm",
+            )
```

Comment:
I think this is missing the migration plugin.

Also you probably need to update the scripts in `tests/scripts/pulp_2to3_migration`. We don't run these as part of the pulp-cli CI since the single container doesn't contain the migration plugin but we do run them in the migration plugin CI.

Reply:
I didn't think we needed a mapping for the migration plugin because unlike the others its now won't change. Let me know if I'm thinking about this right. Here's the status API component listing in a migration box with the "old style" name on it:

```
    "versions": [
        {
            "component": "pulpcore", 
            "version": "3.11.0.dev0"
        }, 
        {
            "component": "pulp_rpm", 
            "version": "3.10.0.dev0"
        }, 
        {
            "component": "pulp_2to3_migration", 
            "version": "0.9.0.dev0"
        }
    ]
```

Similarly it's openAPI schema output (old style):

```
  x-pulp-app-versions:
    pulpcore: 3.11.0.dev0
    pulp_rpm: 3.10.0.dev0
    pulp_2to3_migration: 0.9.0.dev0
```

Here is the name it will use in the future: https://github.com/pulp/pulp-2to3-migration/blob/master/pulp_2to3_migration/app/__init__.py#L8

I don't understand the part about the `deb` aspect. Can you tell me more?

Reply:
It's possible to ship cli plugins separately and there's a pulp-cli-debian plugin at https://github.com/mdellweg/pulp-cli-deb

Reply:
I understand now. I'm adding "deb" because we know about it and it's easy to do here, but also I'm moving this dict out so give plugins the ability to modify this since dicts are imported by reference.

Reply:
Reminder to all:
If we start to support more plugins, and still pulpcore 3.10 (so maybe about half a year from now) we need to add those here, too.